catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/data/impl/MetacardImplTest.java

```diff
@@ -535,4 +535,15 @@ public void testDeserializingHiddenAttribute() throws IOException, ClassNotFound
 
     }
 
+    /*
+     * Test when the resource uri is set with setAttribute() with a URI not a String
+     */
+    @Test
+    public void testSetAttributeResourceUriWithUriParameter() throws URISyntaxException {
+        MetacardImpl metacard = new MetacardImpl();
+        URI uri = new URI(nsUri.toString() + "/resource.html");
+        metacard.setAttribute("resource-uri", uri);
+        assertEquals(uri, metacard.getResourceURI());
+    }
+
 }
```

Comment:
i would add a check with `getAttribute(Metacard.RESOURCE_URI)` and make sure the return is a string